internal/model/account.go

```diff
@@ -27,6 +27,23 @@ type UserConfig struct {
 	MakePublic    bool `json:"MakePublic"`
 }
 
+func (c *UserConfig) Scan(value interface{}) error {
+	switch v := value.(type) {
+	case []byte:
+		json.Unmarshal(v, &c)
+		return nil
+	case string:
+		json.Unmarshal([]byte(v), &c)
+		return nil
+	default:
+		return fmt.Errorf("unsupported type: %T", v)
+	}
+}
+
+func (c UserConfig) Value() (driver.Value, error) {
+	return json.Marshal(c)
+}
+
 // ToDTO converts Account to AccountDTO.
```

Comment:
i just a little confuse about use of this? where we use `AccoundDTO`

Reply:
I'm slowing putting code so we have two different structures to handle data: `model.Account` (and same for others) will have the fields that we store in the database and will be a code representation of what we have in the database, and then the DTO objects (DTO stands for "Data Transfer Object") that will have what we need to work with the data in the code, and probably some helper fields (like `hasArchive`).

The flow would be (example):

```
Database(table=bookmark) -> model.Bookmark -> Bookmark Domain -> model.BookmarkDTO.
```

Right now everything is a DTO object and it's used everywhere, but ideally we would have a struct for database representations, another for DTO, another for HTTP payloads, etc.